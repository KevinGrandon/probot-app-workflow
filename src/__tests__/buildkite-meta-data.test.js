import {Application} from 'probot';
import app from '../release-verification';

import releasePayload from './fixtures/release-payload.json';

jest.mock('node-fetch', () => {
  return jest.fn(() =>
    Promise.resolve({
      json: () => Promise.resolve({web_url: 'fusion-verification-url'}),
    })
  );
});

const mockedFetch = require('node-fetch');

describe('buildkite meta-data', () => {
  let robot;
  let github;

  beforeEach(() => {
    robot = new Application();
    robot.load(app);
    github = {
      issues: {
        createComment: jest.fn(),
      },
      repos: {
        createStatus: jest.fn().mockReturnValue(Promise.resolve(true)),
      },
    };
    // Passes the mocked out GitHub API into out robot instance
    robot.auth = () => Promise.resolve(github);
  });

  it('populates meta-data with branch information', async () => {
    await robot.receive({
      event: 'pull_request',
      payload: releasePayload,
    });
    const jsonData = mockedFetch.mock.calls[0][1].body;
    expect(jsonData).toMatchSnapshot();
  });
});
