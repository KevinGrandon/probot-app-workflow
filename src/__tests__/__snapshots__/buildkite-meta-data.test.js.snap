// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildkite meta-data populates meta-data with branch information 1`] = `"{\\"commit\\":\\"HEAD\\",\\"branch\\":\\"master\\",\\"message\\":\\"probot-playground, Release v1.0.7 - release verification\\",\\"author\\":{\\"name\\":\\"KevinGrandon\\"},\\"meta_data\\":{\\"release-pr-number\\":\\"6\\",\\"release-pr-head-sha\\":\\"23999b445a8daf81238dbcdc08776f4489f4aa98\\",\\"release-pr-head-repo-full-name\\":\\"KevinGrandon/probot-playground\\",\\"release-pr-base-repo-full-name\\":\\"KevinGrandon/probot-playground\\",\\"release-pr-prerelease\\":\\"false\\"}}"`;
